orangecontrib/text/twitter.py

```diff
@@ -0,0 +1,121 @@
+import numpy as np
+import tweepy
+from threading import Lock
+
+from orangecontrib.text.corpus import Corpus
+
+from Orange import data
+
+__all__ = ['Credentials', 'TwitterAPI']
+
+
+class Credentials:
+    """Twitter api credentials."""
+
+    def __init__(self, consumer_key, consumer_secret, callback=None):
+        self.consumer_key = consumer_key
+        self.consumer_secret = consumer_secret
+        self.callback = callback
+        self.auth = tweepy.OAuthHandler(self.consumer_key, self.consumer_secret,
+                                        callback=self.callback)
+
+    def check_credentials(self):
+        try:
+            self.auth.get_authorization_url()
+        except tweepy.TweepError:
+            return False
+
+        return True
+
+    def construct_api(self):
+        return tweepy.API(self.auth)
+
+    def __eq__(self, other):
+        return self.consumer_key == other.consumer_key \
+               and self.consumer_secret == other.consumer_secret
+
+
+class TwitterAPI:
+    """Fetch tweets from the Tweeter API"""
+
+    string_fields = [
+        'text', 'created_at', 'lang'
```

Comment:
@s-alexey Thanks for the suggestion! I am dreaming of including coordinates and one day displaying them in GEOmap, but this is a far-fetched dream since that widget only matches string entries for locations. For now, we could include 'place: country'. What do you think @nikicc?
I would also love to include entities, since it would help with parsing (if that's an option we already have, why not include it directly in the widget?), author_name, favourite_count and in_reply_to_screen_name (which includes the author of the original post if the tweet was a reply).
What do you all think?